ui/components/component-library/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ChangeEvent, KeyboardEvent } from 'react';
+import classnames from 'classnames';
+
+import {
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  IconColor,
+  TextVariant,
+  Display,
+  AlignItems,
+} from '../../../helpers/constants/design-system';
+import type { PolymorphicRef } from '../box';
+
+import { Box, Icon, IconName, Label } from '..';
+
+import { CheckboxProps, CheckboxComponent } from './checkbox.types';
+
+export const Checkbox: CheckboxComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      id,
+      isChecked,
+      isIndeterminate,
+      isDisabled,
+      isReadOnly,
+      isRequired,
+      onChange,
+      className = '',
+      iconProps,
+      title,
+      name,
+      label,
+      labelProps,
+      ...props
+    }: CheckboxProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const handleCheckboxKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter') {
+        onChange?.(event as unknown as ChangeEvent<HTMLInputElement>);
+      }
+    };
+
+    // If no title is provided, use the label as the title only if the label is a string
+    const sanitizedTitle =
+      !title && typeof label === 'string' ? label : title || id;
+
+    return (
+      <Label
```

Comment:
I think I made a bad suggestion because this is still a JS component it may not work. Use Text as="label"